src/Microsoft.Diagnostics.Tools.RuntimeClient/DiagnosticsIpc/IpcHeader.cs

```diff
@@ -0,0 +1,79 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Net.Sockets;
+using System.Runtime.InteropServices;
+using System.Text;
+
+namespace Microsoft.Diagnostics.Tools.RuntimeClient.DiagnosticsIpc
+{
+    public class IpcHeader
+    {
+        IpcHeader() { }
+
+        public IpcHeader(DiagnosticServerCommandSet commandSet, byte commandId)
+        {
+            CommandSet = (byte)commandSet;
+            CommandId = commandId;
+        }
+
+        // the number of bytes for the DiagnosticsIpc::IpcHeader type in native code
+        public static readonly UInt16 HeaderSizeInBytes = 20;
+        private static readonly UInt16 MagicSizeInBytes = 14;
+
+        public byte[] Magic = ASCIIEncoding.ASCII.GetBytes("DOTNET_IPC_V1" + '\0'); // byte[14] in native code
+        public UInt16 Size = HeaderSizeInBytes;
+        public byte CommandSet;
+        public byte CommandId;
+        public UInt16 Reserved = 0x0000;
+
+
+        // Helper expression to quickly get V1 magic string for comparison
+        // should be 14 bytes long
+        public static byte[] DOTNET_IPC_V1 => ASCIIEncoding.ASCII.GetBytes("DOTNET_IPC_V1" + '\0');
```

Comment:
How is this different from Magic?

Reply:
It's a quick little helper expression to get the magic bytes without needing to create an instance of an `IpcHeader`.  It isn't currently used, but if we have more versions in the future, I envisioned having one of these per version for ease of construction since the `Magic` field wouldn't be defaulting to the V1 magic bytes.